src/lib/krb5/ccache/ccselect.c

```diff
@@ -142,32 +144,55 @@ krb5_cc_select(krb5_context context, krb5_principal server,
             return ret;
     }
 
+    ret = krb5_copy_principal(context, server, &srvcp);
```

Comment:
I would prefer to avoid the copy if it's not needed, and set server to point to the copy if we made one.

Also, this PR frees the allocated value in five different exit paths.  Our coding practices state "do not get to the point of freeing the same resource in several different places within a function depending on the exit path."  Introduce a cleanup label and use it.